src/hotspot/share/code/nmethod.hpp

```diff
@@ -249,6 +250,7 @@ class nmethod : public CodeBlob {
 #if INCLUDE_JVMCI
   int      _speculations_offset;
 #endif
+  int      _immutable_data_references_offset;
```

Comment:
Do we really need this? Can we simply use `_immutable_data_end - sizeof(int)` and make sure that `_immutable_data_end` has big enough alignment (which you have already)